### ISSUE TITLE
Overuse of react-native's custom PropType View.propTypes.style

### ISSUE THREAD

**Issue by gcmartins93** [react-native-training/react-native-elements#323]
### TL;DR
Using View.propTypes.style where it's not a View styling object causes warnings when suppling keys that are not applicable to a View.

### Description

At version 0.11, components start using react-native's custom PropType (View.propTypes.style) in almost every possible situations, instead of PropTypes.any - except for fixes made by this b86d6211d264e0495e352525ea4dfd69d73e4562 commit. The problem here is the following:

Those styling's custom PropTypes were made so keys can be validated against applicability to a specific component. Example:

View component has no style color, i.e. we can't apply a color to a View, hence it will throw a warning that it's a invalid key every time we try to do that.

But, what happens if we start handling all component styling as expecting its a View.propTypes.style?

**`Failed prop type: Invalid props.inputStyle key 'color' supplied to 'FormInput'`**

**`Failed prop type: Invalid props.labelStyle key 'color' supplied to 'FormLabel'`**

**`Failed prop type: Invalid props.iconStyle key 'color' supplied to 'Icon'`**

Happens that in examples like inputStyle and labelStyle, they should accept color key as valid, so it's better to define inputStyle as TextInput.propTypes.style and labelStyle as Text.propTypes.style, for example.

**Comment by bozzmob** [comment]
Would like to tag #320 (both are related I believe). Yes, we observe this issue, I think we may have to fix this across components. @Monte9 @binoy14 @tarang9211 thoughts?

**Comment by ChristianMEC** [comment]
I had almost same problem ... on CheckBox at textStyle property need to make a change.

inside of CheckBox.js on 
CheckBox.propTypes ={ 
...
textStyle: **View**.propTypes.style _=> change it to **Text**.propTypes.style_
...
}

**Comment by Monte9** [comment]
@bozzmob definitely related. Anyone wanna submit a PR with fixes? Should be pretty straightforward.

**Comment by bozzmob** [comment]
@Monte9 will take care of it today.

**Comment by binoy14** [comment]
@gcmartins93 good point, and this is my bad. I didn't realize text has few different stylings then view. We definitely need have test coverage around this so it doesn't happen again

**Comment by bozzmob** [comment]
This has been fixed and will be available from v0.11.2
